src/atomate2/vasp/sets/defect.py

```diff
@@ -4,15 +4,39 @@
 
 from pymatgen.core import Structure
 from pymatgen.io.vasp import Outcar, Vasprun
+from pymatgen.io.vasp.inputs import Kpoints, Kpoints_supported_modes
 
 from atomate2.vasp.sets.base import VaspInputGenerator
 
+SPECIAL_KPOINT = Kpoints(
+    comment="special k-point",
+    num_kpts=1,
+    style=Kpoints_supported_modes.Reciprocal,
+    kpts=((0.25, 0.25, 0.25),),
+    kpts_shift=(0, 0, 0),
+    kpts_weights=[1],
+)
+
+SPECIAL_KPOINT_GAMMA = Kpoints(
+    comment="special k-point",
+    num_kpts=1,
+    style=Kpoints_supported_modes.Reciprocal,
+    kpts=((0.25, 0.25, 0.25), (0.0, 0.0, 0.0)),
+    kpts_shift=(0, 0, 0),
+    kpts_weights=[1, 0],
+)
+
 
 @dataclass
-class AtomicRelaxSetGenerator(VaspInputGenerator):
-    """Class to generate VASP atom-only relaxation input sets."""
+class ChargeStateRelaxSetGenerator(VaspInputGenerator):
+    """Generator for atomic-only relaxation for defect supercell calculations.
 
-    use_structure_charge: bool = True
+    Since the defect cells are assumed to be large, we will use only a single k-point.
+    """
+
+    def __post_init__(self):
+        self.use_structure_charge = True
```

Comment:
Please don't use post_init for this! Just use the normal dataclass variable. If people want to override this for any reason they won't be. Similarly for the other places you do this!